src/bigfile.c

```diff
@@ -1678,7 +1676,8 @@ attrset_add_attr(BigAttrSet * attrset, const char * attrname, const char * dtype
     _dtype_normalize(n->dtype, dtype);
 
     n->name = free;
-    strcpy(free, attrname);
+    strncpy(free, attrname, size);
```

Comment:
size is guaranteed to be longer than strlen(attrname) in a few lines above. I think you can use a memcpy with strlen(attrname) + 1, if checkers complained about strcpy.